### ISSUE TITLE
Crash with filter against object since v8.1.1957

### ISSUE THREAD

**Issue by itchyny** [vim/vim#4888]
**Describe the bug**
Using filter function twice against an object causes SIGABRT. See the following script.
```vim
let x = {"x":10}
echo map(range(2), 'filter(copy(x), "1")')
```

**To Reproduce**
Detailed steps to reproduce the behavior:
```
 % cat <<EOF >/tmp/main.vim
> let x = {"x":10}
> echo map(range(2), 'filter(copy(x), "1")')
> EOF
 % cat /tmp/main.vim
let x = {"x":10}
echo map(range(2), 'filter(copy(x), "1")')
 % ./src/vim -u NONE -N -c "source /tmp/main.vim" -c q
Vim: Caught deadly signal ABRT
Vim: Finished.
zsh: abort      ./src/vim -u NONE -N -c "source /tmp/main.vim" -c q
```

**Expected behavior**
No SIGABRT happens.
**Screenshots**
If applicable, copy/paste the text or add screenshots to help explain your problem.

**Environment (please complete the following information):**
 - Vim version v8.1.1957~v8.1.1963 (v8.1.1956 does not reproduce the crash)
 - OS: macOS 10.14
 - Terminal:  iTerm2

**Additional context**

**Comment by brammool** [comment]
I can reproduce it.  filter() and map() are not implemented in a way they can be used recursively.

**Comment by brammool** [comment]
Fixed with Patch 8.1.1964.  Was actually not because of nesting, but because of setting the variable type to string, while the pointer was not NULL.

**Comment by itchyny** [comment]
Thank you for fixing.
